crates/net/network/src/message.rs

```diff
@@ -37,6 +38,52 @@ impl NewBlockMessage {
     }
 }
 
+/// Internal form of a `PooledTransactions` message
+/// Same as [`NewPooledTransactionHashes68`] but types and sizes are Option.
+#[derive(Debug, Clone)]
+pub struct NewPooledTransactionHashes {
+    /// Transaction types for new transactions that have appeared on the network.
+    pub types: Option<Vec<u8>>,
+    /// Transaction sizes for new transactions that have appeared on the network.
+    pub sizes: Option<Vec<usize>>,
+    /// Transaction hashes for new transactions that have appeared on the network.
+    pub hashes: Vec<H256>,
+}
+
+impl From<NewPooledTransactionHashes66> for NewPooledTransactionHashes {
+    fn from(msg: NewPooledTransactionHashes66) -> Self {
+        NewPooledTransactionHashes { types: None, sizes: None, hashes: msg.0 }
+    }
+}
+
+impl From<NewPooledTransactionHashes68> for NewPooledTransactionHashes {
+    fn from(msg: NewPooledTransactionHashes68) -> Self {
+        NewPooledTransactionHashes {
+            types: Some(msg.types),
+            sizes: Some(msg.sizes),
+            hashes: msg.hashes,
+        }
+    }
+}
+
+impl From<(Vec<TxHash>, Vec<TxType>, Vec<usize>)> for NewPooledTransactionHashes {
```

Comment:
From impls for tuples are not great so would abandon them entirely.

but not clear why we need this type at all.

Reply:
follow it up with acb0d81f90e16bdd4c9070208470d7ff0b46b3c5.